### ISSUE TITLE
[Bug][Cosmetics][2.2.0.rc.2] Compilation progress indication is not being displayed cleanly in Windows Command Prompt.

### ISSUE THREAD

**Issue by d-ph** [webpack/webpack#3591]
**Do you want to request a *feature* or report a *bug*?**
A bug.

**If the current behavior is a bug, please provide the steps to reproduce.**
0. You probably need Windows to reproduce it. Problem is reproducible with the Command Prompt and with Cmder.
1. `git clone https://github.com/d-ph/bugreport-webpack-cli-progress-windows.git`
2. `cd bugreport-webpack-cli-progress-windows`
3. `npm install`
4. `npm run build`

Result:
During compilation, the progress indication is not being cleanly displayed. See the following gif:
![bugreport-gif](https://cloud.githubusercontent.com/assets/4347095/21500800/76005b3a-cc3a-11e6-888a-35dc1b171595.gif)

**What is the expected behavior?**
During compilation, the progress indication is being cleanly displayed.

**Please mention other relevant information such as the browser version, Node.js version, Operating System and programming language.**
Windows 10, Ms Command Prompt or Cmder

**Comment by SpaceK33z** [comment]
Thanks for your clear bug report. Would you be interested in doing a PR for this? We can certainly use help for this. The relevant code can [be found here](https://github.com/webpack/webpack/blob/master/lib/ProgressPlugin.js).

**Comment by d-ph** [comment]
I can try. Let me see.

**Comment by d-ph** [comment]
As I mentioned in the PR, the problem isn't solved entirely by that PR. Now the progress indicator stays in one place, but still isn't right:
![bugreport-after-gif](https://cloud.githubusercontent.com/assets/4347095/21523346/4c33e332-cd05-11e6-938f-0a49ba3a904e.gif)

I found, that the problem lies in the fact, that virtually anything can `console.log()` during the progress output, which breaks Windows' terminal. Long story short: This is something nothing can be done about.

But there is a workaround: don't use prescripts in npm. I.e. instead of this:
```json
"scripts": {
    "clean": "rimraf dist",
    "prebuild": "npm run clean",
    "build": "webpack --progress --hide-modules"
  },
```
use this:
```json
"scripts": {
    "clean": "rimraf dist",
    "build": "npm run clean && webpack --progress --hide-modules"
  },
```
which will work in any terminal, any OS. Seems like this is npm's problem and not webpack's.

Bottom line: it's a `Won't Fix`
